fastlane/lib/fastlane/actions/setup_ci.rb

```diff
@@ -7,19 +7,22 @@ def self.run(params)
           return
         end
 
-        case params[:provider]
-        when 'travis'
-          setup_keychain
+        case detect_provider(params)
         when 'circleci'
-          setup_keychain
           setup_output_paths
         end
+
+        setup_keychain
```

Comment:
`setup_keychain` was always called first, this is now called last which seems wrong to me even if probably it does not matter in the case of the additional `setup_output_paths` for Travis. However i believe there is a test which expect a given order and this is breaking.

Reply:
Actually as you commented Message is asserted in reverse order so the test would pass but still I wonder if conceptually the action should not do first what it is supposed to do unless there is a very good reason to do something else first.

Reply:
I actually just followed the suggestion made in your issue #15445:
> If a CI is detected, setup_keychain should be called in all cases **and only after** [...] **more tasks could be executed** with a case/when statement.

So I really don't mind moving this before the `case/when` statement.

Reply:
I meant to say only after calling `setup_keychain`, other actions should be called with...
Sorry for the confusion, I see how this could be interpreted the other way around.